src/lib/simple-server.ts

```diff
@@ -0,0 +1,49 @@
+import { Server } from 'http';
+import express from 'express';
+import { portFinder } from './port-finder';
+
+class SimpleServer {
+	private server: Server | null = null;
+	private port: number | null = null;
+
+	async start( site: SiteDetails ): Promise< number > {
+		console.log( 'site', site );
+		if ( ! site.id || ! site.path ) {
+			throw new Error( 'Site ID is required' );
+		}
+
+		if ( this.server ) {
+			return this.port!;
+		}
+
+		const app = express();
+		this.port = await portFinder.getOpenPort();
+
+		// Add a basic route handler
+		app.get( '/', ( req, res ) => {
```

Comment:
I tried a node-based sql gui module, but it didn't work at all. 

Looking at some static viewers, which we can serve from `app.use( express.static( path.join( __dirname, '../../vendor/sql-viewer/public' ) ) );` or somewhere else.